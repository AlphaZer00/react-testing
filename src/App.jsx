import { useState } from "react";

const App = () => {
    const [heading, setHeading] = useState("Magnificent Monkeys");

    const handleClick = () => {
        setHeading("Radical Rhinos");
    };

    return (
        <>
            <button type="button" onClick={handleClick}>
                Click here
            </button>
            <h1>{heading}</h1>
        </>
    );
};

export default App;
