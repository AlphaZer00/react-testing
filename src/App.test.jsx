import { render, screen } from "@testing-library/react";
import App from "./App";
import { describe, it, expect } from "vitest";

describe("App component", () => {
    it("renders heading correctly", () => {
        render(<App />);
        expect(screen.getByRole("heading").textContent).toMatch(/First Test/i);
    });
});
