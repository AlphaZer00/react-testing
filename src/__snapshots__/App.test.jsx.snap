// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders Magnificent Monk 1`] = `
<div>
  <button
    type="button"
  >
    Click here
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;

exports[`App component > renders Magnificent Monke 1`] = `
<div>
  <button
    type="button"
  >
    Click here
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;

exports[`App component > renders Magnificent Monkey 1`] = `
<div>
  <button
    type="button"
  >
    Click here
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;

exports[`App component > renders Magnificent Monkeys 1`] = `
<div>
  <button
    type="button"
  >
    Click here
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;

exports[`App component > renders Magnificent Monkq 1`] = `
<div>
  <button
    type="button"
  >
    Click here
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;

exports[`App component > renders Magnificent Monkwys 1`] = `
<div>
  <button
    type="button"
  >
    Click here
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;
